src/bootstrap/bootstrap.py

```diff
@@ -508,14 +516,25 @@ def downloading_llvm(self):
         return opt == "true" \
             or (opt == "if-available" and self.build in supported_platforms)
 
+    def get_rustc_commit(self):
+        if self.rustc_commit is None:
+            # Look for a version to compare to based on the current commit.
+            # Only commits merged by bors will have CI artifacts.
+            merge_base = [
+                "git", "rev-list", "--author=bors@rust-lang.org", "-n1",
+                "--merges", "--first-parent", "HEAD"
+            ]
+            self.rustc_commit = subprocess.check_output(merge_base, universal_newlines=True).strip()
+        return self.rustc_commit
+
     def _download_component_helper(
         self, filename, pattern, tarball_suffix, stage0=True, key=None
     ):
         if key is None:
             if stage0:
                 key = self.stage0_compiler.date
             else:
-                key = self.rustc_commit
+                key = self.get_rustc_commit()
```

Comment:
Please don't do this without caching, it will rerun git like 12 times each time you invoke x.py.